src/regtech_api_commons/api/dependencies.py

```diff
@@ -0,0 +1,81 @@
+import os
+from http import HTTPStatus
+from typing import List
+
+import httpx
+from fastapi import Request
+from starlette.authentication import AuthCredentials
+
+from regtech_api_commons.api.exceptions import RegTechHttpException
+from regtech_api_commons.models.auth import AuthenticatedUser
+
+
+def verify_lei(inst_api_url: str):
+    def lei_active_check(request: Request, lei: str) -> None:
+        res = httpx.get(inst_api_url + lei, headers={"authorization": request.headers["authorization"]})
+        lei_obj = res.json()
+        if not lei_obj["is_active"]:
+            raise RegTechHttpException(
+                status_code=HTTPStatus.FORBIDDEN, name="Request Forbidden", detail=f"LEI {lei} is in an inactive state."
+            )
+
+    return lei_active_check
```

Comment:
Just making sure I understand but this is to return a function to be used as a decorator?

Reply:
no, this is so the institution api url can be passed in; so it would return a function that `Depends` can call

Reply:
kinda like a partial